src/main/java/com/snowflake/kafka/connector/internal/SnowflakeSinkServiceV1.java

```diff
@@ -311,6 +329,17 @@ public void setBehaviorOnNullValuesConfig(
     this.behaviorOnNullValues = behavior;
   }
 
+  // enable use of new stage files processor
+  void enableStageFilesProcessor() {
+    this.useStageFilesProcessor = true;
+    if (null != cleanerServiceExecutor) {
+      cleanerServiceExecutor.shutdown();
+    }
+    cleanerServiceExecutor =
+        new ScheduledThreadPoolExecutor(
+            Math.max(1, Runtime.getRuntime().availableProcessors() / 2));
```

Comment:
Why do you need more than 1 thread? It could impact the customer workload and since customers run multiple kafka connect clusters on the same node it could be too much to use the  availableProcessors